src/tests/plans/D3MCompoundPlan.t.sol

```diff
@@ -0,0 +1,346 @@
+
+        uint256 supply = plan.calculateTargetSupply(targetRate);
+        assertEqApproxBPS(supply, _sub(_add(cash, borrows), reserves), 1);
+    }
+
+    function test_interest_rate_calc() public {
+        // Confirm that the inverse function is correct by comparing all percentages
+        for (uint256 i = 1; i <= 100 ; i++) {
+            (uint256 targetRate, uint256 newCash, uint256 borrows, uint256 reserves) = _targetRateForRelativeUtil(i * 100);
+
+            uint256 supply = plan.calculateTargetSupply(targetRate);
+            assertEqApproxBPS(supply, _sub(_add(newCash, borrows), reserves), 1);
+        }
+    }
+
+    function test_calculate_base_rate() public {
+        uint256 supply = plan.calculateTargetSupply(model.baseRatePerBlock());
+        assertEq(supply, 0);
+    }
+
+    function test_calculate_zero_rate() public {
+        uint256 supply = plan.calculateTargetSupply(0);
+        assertEq(supply, 0);
+    }
+
+    function test_very_high_utilization_supported() public {
+        uint256 normalRate = _add(_wmul(model.kink(), model.multiplierPerBlock()), model.baseRatePerBlock());
+
+        uint256 util = 1e36;
+        uint256 rate = normalRate + model.jumpMultiplierPerBlock() * (util - model.kink()) / WAD;
+
+        // Make sure utilization target supply is indeed less than borrows (i.e utilization > 100%)
+        uint256 targetSupply = plan.calculateTargetSupply(rate);
+        assertLt(targetSupply, cDai.totalBorrows());
+
+        // Make sure targetSupply is indeed very small
+        assertGt(targetSupply, 0);
+        assertLt(targetSupply, WAD);
+    }
+
+    function test_implements_getTargetAssets() public override {
+        uint256 initialRatePerBlock = cDai.borrowRatePerBlock();
+
+        plan.file("barb", initialRatePerBlock - (1 * WAD / 1000) / model.blocksPerYear()); // minus 0.1% from current yearly rate
+
+        uint256 initialTargetAssets = plan.getTargetAssets(0);
+        assertGt(initialTargetAssets, 0);
+
+        // Reduce target rate (increase needed number of target Assets)
+        plan.file("barb", initialRatePerBlock - (2 * WAD / 1000) / model.blocksPerYear()); // minus 0.2% from current yearly rate
+
+        uint256 newTargetAssets = plan.getTargetAssets(0);
+        assertGt(newTargetAssets, initialTargetAssets);
+    }
+
+    function test_getTargetAssets_barb_zero() public {
+        assertEq(plan.barb(), 0);
+        assertEq(plan.getTargetAssets(0), 0);
+    }
+
+    function test_getTargetAssets_current_rate() public {
+        cDai.accrueInterest();
+        uint256 borrowRatePerBlock = cDai.borrowRatePerBlock();
+        plan.file("barb", borrowRatePerBlock);
+
+        uint256 targetAssets = plan.getTargetAssets(0);
+        assertEqAbsolute(targetAssets, 0, WAD);
+    }
+
+    function test_rate_model_changed_not_active() public {
+        // Simulate Compound changing the rate model in the pool
```

Comment:
```suggestion
        // Simulate Compound changing the rate model in the pool
        assertTrue(address(plan.tack()) == cDai.interestRateModel());
        assertTrue(plan.active());
```

Reply:
Actually we also need to set barb so it doesn't fail due that.

Reply:
Handled in `eba23f7018`